### PR TITLE
1817 review tab against checklist

### PR DESCRIPTION
**Pull request by GCHQ-Developer-299**
## Summary of the changes
- Add dark mode CSS in prep for that
- Made tab labels no-wrap
- Added min-max content story and visual regression test
- High contrast visual regression tests

## Related issue
#1817

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

File: packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx [hunk 1]
```diff
@@ -566,4 +568,58 @@ describe("IcTab visual regression tests", () => {
       testThreshold: DEFAULT_TEST_THRESHOLD + 0.012,
     });
   });
+
+  it("renders min and max content tabs", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx:572]:
```suggestion
  it("should render min and max content tabs", () => {
```

File: packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx [hunk 2]
```diff
@@ -566,4 +568,58 @@ describe("IcTab visual regression tests", () => {
...
+
+  after(() => {
+    cy.disableForcedColors();
+  });
+
+  afterEach(() => {
+    cy.task("generateReport");
+  });
+
+  it("renders a tab group with tab focussed", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx:595]:
```suggestion
  it("should render a tab group with tab focussed", () => {
```

File: packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx [hunk 3]
```diff
@@ -566,4 +568,58 @@ describe("IcTab visual regression tests", () => {
...
+    cy.focused().as("activeElement");
+    cy.get("@activeElement").should(CONTAIN_TEXT, "Method");
+
+    cy.compareSnapshot({
+      name: "focused_tabs_high_contrast",
+      testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.014),
+    });
+  });
+
+  it("renders disabled tab", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx:608]:
```suggestion
  it("should render disabled tab", () => {
```

File: packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx [hunk 4]
```diff
@@ -566,4 +568,58 @@ describe("IcTab visual regression tests", () => {
...
+    mount(<DisabledTab />);
+    cy.checkA11yWithWait();
+    cy.compareSnapshot({
+      name: "disabled_tab_high_contrast",
+      testThreshold: DEFAULT_TEST_THRESHOLD + 0.011,
+    });
+  });
+
+  it("renders inline tab group", () => {
```
Comment (GCHQ-Developer-465) [inline packages/react/src/component-tests/IcTabs/IcTabs.cy.tsx:617]:
```suggestion
  it("should render inline tab group", () => {
```

**Review by GCHQ-Developer-465 — CHANGES_REQUESTED** [review]
For consistency reasons and due to my upcoming pr please update the test phrasing 😊

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**